src/ts/ts-genapi/export.ts

```diff
@@ -0,0 +1,356 @@
+    Navigation: IApiViewNavItem[];
+}
+
+declare interface IApiViewToken {
+    Kind: ApiViewTokenKind;
+    DefinitionId?: string;
+    NavigateToId?: string;
+    Value?: string;
+}
+
+declare interface IApiViewNavItem {
+    Text: string;
+    NavigationId: string;
+    ChildItems: IApiViewNavItem[];
+    Tags: {
+        [propertyName: string]: string;
+    };
+}
+
+function appendMembers(builder: TokensBuilder, navigation: IApiViewNavItem[], item: ApiItem)
+{
+    builder.lineId(item.canonicalReference.toString());
+    builder.indent();
+    if (item instanceof ApiDeclaredItem) {
+        item.excerptTokens.forEach(token => {
+            if (token.kind == ExcerptTokenKind.Reference)
+            {
+                builder.typeReference(token.canonicalReference.toString(), token.text);
+            }
+            else
+            {
+                builder.splitAppend(token.text, item.canonicalReference.toString(), item.displayName);
+            }
+        });
+    }
+
+    var navigationItem: IApiViewNavItem;
+    var typeKind: string;
+
+    switch (item.kind)
+    {
+        case ApiItemKind.Interface:
+        case ApiItemKind.Class:
+            typeKind = item.kind.toLowerCase();
+            break
+        case ApiItemKind.TypeAlias:
+            typeKind = "struct";
+            break
+    }
+
+    if (typeKind != null)
+    {
+        navigationItem = {
+            Text: item.displayName,
+            NavigationId: item.canonicalReference.toString(),
+            Tags: {
+                "TypeKind": typeKind
+            },
+            ChildItems: []
+        };
+        navigation.push(navigationItem);
+    }
+
+    if (item.kind == ApiItemKind.Interface ||
+        item.kind == ApiItemKind.Class)
+    {
+        if (item.members.length > 0)
+        {
+            builder
+                .punct("{")
+                .newline()
+                .incIndent()
+    
+            item.members.forEach(member => {
+                appendMembers(builder,navigationItem.ChildItems, member);
+            });
+
+            builder
+                .decIndent()
+                .indent()
+                .punct("}")
+                .newline();
+        }
+        else
+        {
+            builder
+                .punct("{")
+                .space()
+                .punct("}")
+                .newline();
+        }
+    }
+    else
+    { 
+        builder.newline();
+    }
+}
+
+const apiModel: ApiModel = new ApiModel();
+
+apiModel.loadPackage(process.argv[2]);
+
+var navigation: IApiViewNavItem[] = [];
+var builder: TokensBuilder = new TokensBuilder();
+
+apiModel.packages.forEach(p => 
+    p.entryPoints.forEach(e => 
+        e.members.forEach(m => appendMembers(builder, navigation, m))
+        )
+);
+
+var apiViewFile: IApiViewFile = {
+    Name: apiModel.packages[0].name,
+    Navigation: navigation,
+    Tokens: builder.tokens
+}
+
+writeFile(process.argv[3], JSON.stringify(apiViewFile), err => {
```

Comment:
if you want to use async/await instead of callbacks: https://masteringjs.io/tutorials/node/promisify